src/views/Characters/Characters.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { fetchCharacters } from '../../services/characters';
+import CharactersList from '../../components/Characters/CharactersList';
+import Controls from '../../components/Controls/Controls';
+
+export default function Characters() {
+  const [character, setCharacter] = useState([]);
+  const [query, setQuery] = useState('');
+  const [race, setRace] = useState('All');
+
+  useEffect(() => {
+    const getCharacter = async () => {
+      const response = await fetchCharacters(race, query);
+      setCharacter(response);
+    };
+    getCharacter();
+  }, [race, query]);
```

Comment:
Adding query into the dependency array here is causing react to fetch new data each time the query is changed.